### PR TITLE
webgui: introduce web window manager

### PR DESCRIPTION
**Pull request by linev**
This is central instance which will handle all kinds of displays, implemented with web

This will be TCanvas, TFitPanel, TBrowser, TGEditor and any kind of user-specific UIs.

TWebWindow manage all kinds websocket-like connections and allow multiple connections for each window - means many clients can be opened for each existing canvas or browser (if allowed).

Also all kind of web front-end are managed by the window manager - CEF3, Qt5 WebEngine or normal web browser.

JSROOT provides special handle to communicate with web window via all supported channels. Any custom HTML widget can directly use such handle and create absolutely custom UI without use of openui5 or largest part of JSROOT itself.

First implementation of v7 th1/th2 histograms and canvas drawing is provided

Several fixes in v7 classes concerning I/O, converting into v6 histograms for drawing is disabled, code can be removed later.

@bellenot 
Travis CI will be unhappy with code formatting - please keep it as is for the moment, one could reformat complete code later

### CODE REVIEW

File: net/http/inc/THttpWSHandler.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+class THttpWSHandler : public TNamed {
+
+friend class THttpServer;
+
+private:
+
+   THttpWSEngine *FindEngine(UInt_t id) const;
+
+   Bool_t DirecltyHandle(THttpCallArg *arg);
```
Comment (bellenot) [inline net/http/inc/THttpWSHandler.h]: Shouldn't it be DirectlyHandle? (BTW, I don't really like this name...)

File: gui/canvaspainter/v7/src/TCanvasPainter.cxx [hunk 2]
```diff
@@ -198,254 +195,245 @@ class TCanvasPainter : public THttpWSHandler,
...
+struct TNewCanvasPainterReg {
+   TNewCanvasPainterReg() { TCanvasPainter::GeneratorImpl::SetGlobalPainter(); }
+   ~TNewCanvasPainterReg() { TCanvasPainter::GeneratorImpl::ResetGlobalPainter(); }
+} newCanvasPainterReg;
 
-/// \}
 
-} // unnamed namespace
+} // namespace Experimental
+} // namespace ROOT
```
Comment (bellenot) [inline gui/canvaspainter/v7/src/TCanvasPainter.cxx]: Couldn't you keep the namespaces until the end? That would save you to use long names like: `ROOT::Experimental::TCanvasPainter` every time

File: net/http/inc/THttpWSHandler.h (hunk 1 above)
Comment (linev) [reply net/http/inc/THttpWSHandler.h]: Yes, you are right - it was misspelling of DirectlyHandle. Now I call it HandleWS

Comment (linev) [reply net/http/inc/THttpWSHandler.h]: And anyway - it is private function, which is not accessible by the end-user

Comment (bellenot) [reply net/http/inc/THttpWSHandler.h]: OK, thanks for changing it (and private or not, I just didn't like it...) :-)

File: gui/canvaspainter/v7/src/TCanvasPainter.cxx (hunk 2 above)
Comment (linev) [reply gui/canvaspainter/v7/src/TCanvasPainter.cxx]: It is like @Axel-Naumann is doing in all other parts of v7 code.
On the long run `::Experimental` namespace will disappear and name `ROOT::TCanvasPainter` should be ok.

Comment (bellenot) [reply gui/canvaspainter/v7/src/TCanvasPainter.cxx]: OK, fine with me then

**Pull request merged**